Woof/WoofTests/ViewModel/LoginViewModelTests/LoginViewModelSetUserRoleTests.swift

```diff
@@ -0,0 +1,71 @@
+import XCTest
+
+final class LoginViewModelSetUserRoleTests: XCTestCase {
+    private var userPreferencesStorage = KeyValueStorage.testPreferencesStorage
+    private let testLoginViewModel = LoginViewModel()
+
+    override func setUp() {
+        super.setUp()
+        userPreferencesStorage.deleteData(for: KeyValueStorage.Key.userPreferences)
+    }
+
+    func testMethodExistsInAPI() {
+        testLoginViewModel.set(.sitter)
+    }
+
+    func testSuccessfullyOverwritesTheDefaultValueForTheOwnerRole() {
+        // Given
+        let initialRole = Role.none
+        let overriddenRole = Role.owner
+        testLoginViewModel.set(initialRole)
+
+        // When
+        testLoginViewModel.set(overriddenRole)
+
+        // Then
+        XCTAssertNotEqual(PreferencesHandler.getUserRole(), initialRole)
+        XCTAssertEqual(PreferencesHandler.getUserRole(), overriddenRole)
+    }
+
+    func testSuccessfullyOverwritesTheDefaultValueForTheSitterRole() {
+        // Given
+        let initialRole = Role.none
+        let overriddenRole = Role.sitter
+        testLoginViewModel.set(initialRole)
+
+        // When
+        testLoginViewModel.set(overriddenRole)
+
+        // Then
+        XCTAssertNotEqual(PreferencesHandler.getUserRole(), initialRole)
+        XCTAssertEqual(PreferencesHandler.getUserRole(), overriddenRole)
+    }
+
+    func testSuccessfullyOverwritesTheSitterRoleValueForDefault() {
```

Comment:
nit:

keep it short:
```suggestion
    func testOverridesSavedSitterRoleValueWithNoneRoleValue() {
```

Reply:
removed this test